src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3138 @@
+		newKnobPos =
+		    firstPadValue
+		    + (xDisplay - firstPadX) * ((((secondPadValue - firstPadValue) << 20) / (secondPadX - firstPadX)) >> 20);
+	}
+
+	//in the deluge knob positions are stored in the range of -64 to + 64, so need to adjust newKnobPos set above.
+	newKnobPos = newKnobPos - kKnobPosOffset;
+
+	return newKnobPos;
+}
+
+//used to calculate new knobPos when you turn the mod encoders (gold knobs)
+int32_t AutomationInstrumentClipView::calculateKnobPosForModEncoderTurn(int32_t knobPos, int32_t offset) {
+
+	//adjust the current knob so that it is within the range of 0-127 for calculation purposes
+	knobPos = knobPos + kKnobPosOffset;
+
+	int32_t newKnobPos = 0;
+
+	if ((knobPos + offset) < 0) {
+		newKnobPos = knobPos;
+	}
+	else if ((knobPos + offset) <= kMaxKnobPos) {
+		newKnobPos = knobPos + offset;
+	}
+	else if ((knobPos + offset) > kMaxKnobPos) {
+		newKnobPos = kMaxKnobPos;
+	}
+	else {
+		newKnobPos = knobPos;
+	}
+
+	//in the deluge knob positions are stored in the range of -64 to + 64, so need to adjust newKnobPos set above.
+	newKnobPos = newKnobPos - kKnobPosOffset;
+
+	return newKnobPos;
+}
+
+//used to disable certain actions on the automation overview screen
+//e.g. doubling clip length, editing clip length
+bool AutomationInstrumentClipView::isOnParameterGridMenuView() {
+
+	InstrumentClip* clip = getCurrentClip();
+
+	if (clip->lastSelectedParamID == kNoLastSelectedParamID) {
+		return true;
+	}
+	return false;
+}
+
+//displays patched param names or midi cc names
+void AutomationInstrumentClipView::displayParameterName(int32_t paramID) {
+
+	InstrumentClip* clip = getCurrentClip();
+	Instrument* instrument = (Instrument*)clip->output;
+	char modelStackMemory[MODEL_STACK_MAX_SIZE];
+	ModelStackWithTimelineCounter* modelStack = currentSong->setupModelStackWithCurrentClip(modelStackMemory);
+	ModelStackWithAutoParam* modelStackWithParam =
+	    getModelStackWithParam(modelStack, clip, paramID, clip->lastSelectedParamType);
+	bool isAutomated = false;
+
+	//check if Parameter is currently automated so that the automation status can be drawn on the screen with the Parameter Name
+	if (modelStackWithParam && modelStackWithParam->autoParam) {
+		if (modelStackWithParam->autoParam->isAutomated()) {
+			isAutomated = true;
+		}
+	}
+
+	if (instrument->type == InstrumentType::SYNTH || instrument->type == InstrumentType::KIT) {
+
+		char buffer[30];
+
+		//drawing Parameter Names on 7SEG isn't legible and not done currently, so won't do it here either
+#if HAVE_OLED
+
+		if (clip->lastSelectedParamType == PATCHED) {
+			strcpy(buffer, getPatchedParamDisplayNameForOLED(paramID));
+		}
+		else if (clip->lastSelectedParamType == UNPATCHED) {
+			strcpy(buffer, getUnpatchedParamDisplayNameForOLED(paramID));
+		}
+		else if (clip->lastSelectedParamType == GLOBAL_EFFECTABLE) {
+			strcpy(buffer, getGlobalEffectableParamDisplayNameForOLED(paramID));
+		}
+
+		if (isAutomated) {
+			strcat(buffer, "\n(automated)");
```

Comment:
`strncat`

Reply:
strncat gives me an invalid arguments error:

Invalid arguments '
Candidates are:
char * strncat(char *, const char *, unsigned int)
'

Reply:
right, you need to provide the size of the buffer as the final argument (it has a different signature and prevents buffer overruns)

Reply:
Works now but the compiler gives this warning. Do you know what it means? Thx

![image](https://github.com/SynthstromAudible/DelugeFirmware/assets/138174805/ba6bec15-fd0a-453b-87a1-78fc4587c7ff)

Reply:
30 should be 29 (limit does not include terminating null for whatever reason)